port/osx/omrintrospect.c

```diff
@@ -274,21 +303,17 @@ setupNativeThread(J9ThreadWalkState *state, thread_context *sigContext)
 	return rc;
 }
 
-static int
+static int32_t
 getThreadContext(J9ThreadWalkState *state)
 {
-	int ret = 0;
-	char buffer[1];
+	int32_t ret = 0;
 	PlatformWalkData *data = (PlatformWalkData *)state->platform_data;
 	thread_port_t thread = data->threadList[data->threadIndex];
+	pthread_t pthread = NULL;
 
-	/* Create a pipe to allow the resumed thread to report it has completed
-	 * sending its context info.
-	 */
-	ret = pipe(pipeFileDescriptor);
-
-	if (0 == ret) {
-		ret = pthread_kill(pthread_from_mach_thread_np(thread), SUSPEND_SIG);
+	pthread = pthread_from_mach_thread_np(thread);
+	if (NULL != pthread) {
+		ret = pthread_kill(pthread, SUSPEND_SIG);
 	}
```

Comment:
Please move the call to `pthread_from_mach_thread_np()` into the initialization of `pthread`.
I think `ret` should be something other than `0` if `NULL == pthread`.